src/rpc/coinjoin.cpp

```diff
@@ -45,49 +81,99 @@ static RPCHelpMan coinjoin()
     }
 
+    ValidateCoinJoinArguments();
+
     auto cj_clientman = node.coinjoin_loader->walletman().Get(wallet->GetName());
     CHECK_NONFATAL(cj_clientman != nullptr);
 
-    if (request.params[0].get_str() == "start") {
-        {
-            LOCK(wallet->cs_wallet);
-            if (wallet->IsLocked(true))
-                throw JSONRPCError(RPC_WALLET_UNLOCK_NEEDED, "Error: Please unlock wallet for mixing with walletpassphrase first.");
-        }
-
-        if (!cj_clientman->StartMixing()) {
-            throw JSONRPCError(RPC_INTERNAL_ERROR, "Mixing has been started already.");
-        }
-
-        const ChainstateManager& chainman = EnsureChainman(node);
-        CTxMemPool& mempool = EnsureMemPool(node);
-        CConnman& connman = EnsureConnman(node);
-        bool result = cj_clientman->DoAutomaticDenominating(chainman.ActiveChainstate(), connman, mempool);
-        return "Mixing " + (result ? "started successfully" : ("start failed: " + cj_clientman->GetStatuses().original + ", will retry"));
+    {
+        LOCK(wallet->cs_wallet);
+        if (wallet->IsLocked(true))
+            throw JSONRPCError(RPC_WALLET_UNLOCK_NEEDED, "Error: Please unlock wallet for mixing with walletpassphrase first.");
+    }
+
+    if (!cj_clientman->StartMixing()) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Mixing has been started already.");
     }
 
-    if (request.params[0].get_str() == "stop") {
-        cj_clientman->StopMixing();
-        return "Mixing was stopped";
+    const ChainstateManager& chainman = EnsureChainman(node);
+    CTxMemPool& mempool = EnsureMemPool(node);
+    CConnman& connman = EnsureConnman(node);
+    bool result = cj_clientman->DoAutomaticDenominating(chainman.ActiveChainstate(), connman, mempool);
+    return "Mixing " + (result ? "started successfully" : ("start failed: " + cj_clientman->GetStatuses().original + ", will retry"));
+},
+    };
+}
+
+static RPCHelpMan coinjoin_stop()
+{
+    return RPCHelpMan{"coinjoin stop",
+        "\nStop CoinJoin mixing\n",
+        {},
+        RPCResult{
+            RPCResult::Type::STR, "", "Status of request"
+        },
+        RPCExamples{
+            HelpExampleCli("coinjoin stop", "")
+          + HelpExampleRpc("coinjoin stop", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    if (!wallet) return NullUniValue;
+
+    const NodeContext& node = EnsureAnyNodeContext(request.context);
+
+    if (node.mn_activeman) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Client-side mixing is not supported on masternodes");
     }
 
-    if (request.params[0].get_str() == "reset") {
-        cj_clientman->ResetPool();
-        return "Mixing was reset";
+    ValidateCoinJoinArguments();
+
+    auto cj_clientman = node.coinjoin_loader->walletman().Get(wallet->GetName());
```

Comment:
Could you add also CHECK_NONFATAL that node.coinjoin_loader is not null?

Reply:
Resolved in latest push!